src/protocols/rdp/rdp.c

```diff
@@ -115,7 +115,6 @@ BOOL rdp_freerdp_pre_connect(freerdp* instance) {
 
         /* Downgrade the lock to allow for concurrent read access */
         guac_rwlock_release_lock(&(rdp_client->lock));
-        guac_rwlock_acquire_read_lock(&(rdp_client->lock));
```

Comment:
Isn't this going to just result in no lock being held at all? It sure looks like the read lock is still needed here.

Reply:
If I understand the way the locks work the read lock still exists and releasing the write lock switches back to using the read lock.

Reply:
Ah, right you are. Ignore me.